vault/expiration_test.go

```diff
@@ -158,8 +158,11 @@ func TestExpiration_RegisterAuth_NoLease(t *testing.T) {
 	}
 
 	// Should not be able to renew, no expiration
-	_, err = exp.RenewToken(&logical.Request{}, "auth/github/login", root.ID, 0)
-	if err.Error() != "lease not found or lease is not renewable" {
+	resp, err := exp.RenewToken(&logical.Request{}, "auth/github/login", root.ID, 0)
+	if resp == nil {
```

Comment:
Can we check for `err != ErrInvalidRequest` before the `resp` check. This way, if a nil response is returned, there may be chances that the reason for it is already contained in the `err`.

How about `if err != ErrInvalidRequest || resp == nil || resp.Error().Error() != "..."`, print both `resp` and `err`?
I also wonder if we fail when the `err == nil`??